### PR TITLE
VerneMQ session tracing

### PR DESCRIPTION
**Pull request by larshesel**
If you wanna try @ioolkos :)

To get started run `vmq-admin trace` and see where that takes you...

### CODE REVIEW

**Comment by larshesel** [comment]
Need to figure how to handle/not handle mountpoints. Also this currently will only trace a single session and just chooses one if there are multiple when starting the trace and when a new session is created that one will be traced.

File: apps/vmq_server/src/vmq_tracer.erl [hunk 1]
```diff
@@ -0,0 +1,472 @@
+-module(vmq_tracer).
```
Comment (dergraf) [inline apps/vmq_server/src/vmq_tracer.erl]: Add Copyright

Comment (larshesel) [inline apps/vmq_server/src/vmq_tracer.erl]: Done

**Comment by larshesel** [comment]
@dergraf @ioolkos I've cleaned up the code and added more documentation (also a bit on the `vmq-admin` side to make it clear the tracing sub-system cannot be considered stable in features and output). But this is my candidate for merging if we want it to ship with 1.0 - I'm sure there are a *lot* of things that could be much better than they are now :)

**Comment by larshesel** [comment]
@drf maybe this would also be of interest to you to play with - it's likely it'll go into the 1.0 release which is imminent.

**Comment by dergraf** [comment]
Looks freaking awesome!

**Pull request merged**